client/signup/wpcom-login-form/index.jsx

```diff
@@ -1,63 +1,46 @@
 import config from '@automattic/calypso-config';
-import debugFactory from 'debug';
-import { Component } from 'react';
-
-const debug = debugFactory( 'calypso:signup:wpcom-login' );
-
-export default class WpcomLoginForm extends Component {
-	form = null;
-
-	componentDidMount() {
-		debug( 'submit form' );
-		this.form.submit();
+import { useEffect, useRef } from 'react';
+
+function getFormAction( redirectTo ) {
+	const subdomainRegExp = /^https?:\/\/([a-z0-9]+)\.wordpress\.com(?:$|\/)/;
+	const hostname = config( 'hostname' );
+	let subdomain = '';
+
+	if (
+		subdomainRegExp.test( redirectTo ) &&
+		hostname !== 'wpcalypso.wordpress.com' &&
+		hostname !== 'horizon.wordpress.com'
+	) {
+		subdomain = redirectTo.match( subdomainRegExp )[ 1 ] + '.';
 	}
 
-	action() {
-		const subdomainRegExp = /^https?:\/\/([a-z0-9]*).wordpress.com/;
-		let subdomain = '';
-
-		if (
-			subdomainRegExp.test( this.props.redirectTo ) &&
-			config( 'hostname' ) !== 'wpcalypso.wordpress.com' &&
-			config( 'hostname' ) !== 'horizon.wordpress.com'
-		) {
-			subdomain = this.props.redirectTo.match( subdomainRegExp )[ 1 ] + '.';
-		}
+	return `https://${ subdomain }wordpress.com/wp-login.php`;
+}
 
-		return `https://${ subdomain }wordpress.com/wp-login.php`;
-	}
+export default function WpcomLoginForm( { extraFields, redirectTo, authorization, pwd, log } ) {
+	const form = useRef();
 
-	renderExtraFields() {
-		const { extraFields } = this.props;
+	useEffect( () => {
+		form.current.submit();
+	}, [] );
 
+	function renderExtraFields() {
 		if ( ! extraFields ) {
 			return null;
 		}
 
-		return (
-			<div>
-				{ Object.keys( extraFields ).map( ( field ) => {
-					return (
-						<input key={ field } type="hidden" name={ field } value={ extraFields[ field ] } />
-					);
-				} ) }
-			</div>
-		);
+		return Object.entries( extraFields ).map( ( [ field, value ] ) => (
+			<input key={ field } type="hidden" name={ field } value={ value } />
+		) );
```

Comment:
I would personally inline this into one big JSX chunk but extra function is OK, too.

Reply:
Did that with https://github.com/Automattic/wp-calypso/pull/68099/commits/ab227691f39ef17f3c9fe23041530b3facf9b722